packages/query-core/src/query.ts

```diff
@@ -355,7 +350,7 @@ export class Query<
 
     if (!Array.isArray(this.options.queryKey)) {
       if (process.env.NODE_ENV !== 'production') {
-        this.logger.error(
+        console.error(
           `As of v4, queryKey needs to be an Array. If you are using a string like 'repoData', please change it to an Array, e.g. ['repoData']`,
         )
       }
```

Comment:
just seeing this coincidentally: Can we move the `!Array.isArray` check inside the `process.env` check so that it can be reliably removed by bundlers. Maybe it works anyways because the if statement is empty, but I'd still rather have the order switched.